### ISSUE TITLE
Packer amazon-ebs owners array getting corrupted

### ISSUE THREAD

**Issue by finchr** [hashicorp/packer#8491]
#### Overview of the Issue

It appears the owners array in the amazon-ebs builder is getting corrupted:

````
==> Some builds didn't complete successfully and had errors:
--> amazon-ebs: No AMI was found matching filters: {
  Filters: [{
      Name: "name",
      Values: ["amzn2-ami-hvm-2.0*"]
    },{
      Name: "root-device-type",
      Values: ["ebs"]
    },{
      Name: "virtualization-type",
      Values: ["hvm"]
    }],
  Owners: ["774391758379","774391758379","774391758379"]
}
````
Manually setting the desired owner id to the end of the list fixes the issue. 

#### Reproduction Steps

Steps to reproduce this issue

### Packer version

1.4.5

### Simplified Packer Buildfile

````
      "source_ami_filter": {
        "filters": {
          "virtualization-type": "hvm",
          "name": "{{user `aws_ami_filter`}}*",
          "root-device-type": "ebs"
        },
        "owners": [
          "amazon",
          "774391758379",
          "383156758163"
        ],
        "most_recent": true
      },
````

### Operating system and Environment details

Jenkins 2.190.3 running in Kubernetes v1.12.8.

I can also reproduce this running packer directly on my Macbook.

### Log Fragments and crash.log files

````
Started by user admin
Obtained ami/Jenkinsfile from git git@github.com:Infoblox-CTO/Devops.git
Running in Durability level: MAX_SURVIVABILITY
[Pipeline] Start of Pipeline
[Pipeline] podTemplate
[Pipeline] {
[Pipeline] node
Still waiting to schedule task
‘testamibuild-27-v21xd-550wc-xm2mt’ is offline
Agent testamibuild-27-v21xd-550wc-xm2mt is provisioned from template Kubernetes Pod Template
---
apiVersion: "v1"
kind: "Pod"
metadata:
  annotations:
    buildUrl: "http://jenkins:8080/jenkins/job/TestAMIBuild/27/"
  labels:
    some-label: "some-label-value"
    namespace: "jenkins"
    jenkins: "slave"
    jenkins/TestAMIBuild_27-v21xd: "true"
  name: "testamibuild-27-v21xd-550wc-xm2mt"
spec:
  containers:
  - command:
    - "cat"
    image: "044584128299.dkr.ecr.us-gov-west-1.amazonaws.com/athena-devops/devops-admin:201912.15.0"
    name: "packer"
    tty: true
    volumeMounts:
    - mountPath: "/home/jenkins/agent"
      name: "workspace-volume"
      readOnly: false
  - env:
    - name: "JENKINS_SECRET"
      value: "********"
    - name: "JENKINS_TUNNEL"
      value: "jenkins-agent:50000"
    - name: "JENKINS_AGENT_NAME"
      value: "testamibuild-27-v21xd-550wc-xm2mt"
    - name: "JENKINS_NAME"
      value: "testamibuild-27-v21xd-550wc-xm2mt"
    - name: "JENKINS_AGENT_WORKDIR"
      value: "/home/jenkins/agent"
    - name: "JENKINS_URL"
      value: "http://jenkins:8080/jenkins/"
    image: "jenkins/jnlp-slave:3.35-5-alpine"
    name: "jnlp"
    volumeMounts:
    - mountPath: "/home/jenkins/agent"
      name: "workspace-volume"
      readOnly: false
  nodeSelector:
    beta.kubernetes.io/os: "linux"
  restartPolicy: "Never"
  securityContext: {}
  volumes:
  - emptyDir:
      medium: ""
    name: "workspace-volume"

Running on testamibuild-27-v21xd-550wc-xm2mt in /home/jenkins/agent/workspace/TestAMIBuild
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Declarative: Checkout SCM)
[Pipeline] checkout
using credential 7700aaa8-9efb-48be-9e62-65b5cda6c868
Cloning the remote Git repository
Cloning repository git@github.com:Infoblox-CTO/Devops.git
 > git init /home/jenkins/agent/workspace/TestAMIBuild # timeout=10
Fetching upstream changes from git@github.com:Infoblox-CTO/Devops.git
 > git --version # timeout=10
using GIT_SSH to set credentials 
 > git fetch --tags --force --progress -- git@github.com:Infoblox-CTO/Devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
Checking out Revision 37911d9a17607b3204dc62ec1decb56e152b7293 (refs/remotes/origin/rf-test-ami-build)
 > git config remote.origin.url git@github.com:Infoblox-CTO/Devops.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url git@github.com:Infoblox-CTO/Devops.git # timeout=10
Fetching upstream changes from git@github.com:Infoblox-CTO/Devops.git
using GIT_SSH to set credentials 
 > git fetch --tags --force --progress -- git@github.com:Infoblox-CTO/Devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse refs/remotes/origin/rf-test-ami-build^{commit} # timeout=10
 > git rev-parse refs/remotes/origin/origin/rf-test-ami-build^{commit} # timeout=10
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 37911d9a17607b3204dc62ec1decb56e152b7293 # timeout=10
Commit message: "fix missing ssh dependency for docker"
 > git rev-list --no-walk 37911d9a17607b3204dc62ec1decb56e152b7293 # timeout=10
[Pipeline] }
[Pipeline] // stage
[Pipeline] withEnv
[Pipeline] {
[Pipeline] withEnv
[Pipeline] {
[Pipeline] stage
[Pipeline] { (build)
[Pipeline] container
[Pipeline] {
[Pipeline] withEnv
[Pipeline] {
[Pipeline] sh
+ '[' -d ami ']'
+ cd ami
+ make devops-admin
Building amazon-ebs image in region  using profile gov-preprod
packer build --only amazon-ebs --force \
  --var-file=vars/us-gov-west-1.json \
  --var-file=vars/devops-admin.json \
  packer_templates/sdp-ansible.json
[1;32mamazon-ebs output will be in this color.[0m

[1;32m==> amazon-ebs: Force Deregister flag found, skipping prevalidating AMI Name[0m
[1;31m==> amazon-ebs: No AMI was found matching filters: {
==> amazon-ebs:   Filters: [{
==> amazon-ebs:       Name: "name",
==> amazon-ebs:       Values: ["amzn2-ami-hvm-2.0*"]
==> amazon-ebs:     },{
==> amazon-ebs:       Name: "root-device-type",
==> amazon-ebs:       Values: ["ebs"]
==> amazon-ebs:     },{
==> amazon-ebs:       Name: "virtualization-type",
==> amazon-ebs:       Values: ["hvm"]
==> amazon-ebs:     }],
==> amazon-ebs:   Owners: ["774391758379","774391758379","774391758379"]
==> amazon-ebs: }[0m
[1;31mBuild 'amazon-ebs' errored: No AMI was found matching filters: {
  Filters: [{
      Name: "name",
      Values: ["amzn2-ami-hvm-2.0*"]
    },{
      Name: "root-device-type",
      Values: ["ebs"]
    },{
      Name: "virtualization-type",
      Values: ["hvm"]
    }],
  Owners: ["774391758379","774391758379","774391758379"]
}[0m

==> Some builds didn't complete successfully and had errors:
--> amazon-ebs: No AMI was found matching filters: {
  Filters: [{
      Name: "name",
      Values: ["amzn2-ami-hvm-2.0*"]
    },{
      Name: "root-device-type",
      Values: ["ebs"]
    },{
      Name: "virtualization-type",
      Values: ["hvm"]
    }],
  Owners: ["774391758379","774391758379","774391758379"]
}

==> Builds finished but no artifacts were created.
make: *** [vars/devops-admin.json] Error 1
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // container
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // podTemplate
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE
````

**Comment by azr** [comment]
Hey @finchr, thanks for opening; I think this was fixed in #8417, the fix was released in Packer 1.5 yesterday so updating should solve your problem !

**Comment by finchr** [comment]
Thanks @azr!
